### ISSUE TITLE
Windows PyPy fails to find an interpreter

### ISSUE THREAD

**Issue by henryiii** [PyO3/maturin-action#267]
Windows 64-bit PyPy fails to find an interpreter - Linux and macOS (both archs) are fine. Example: https://github.com/tox-dev/pyproject-fmt-rust/pull/11

It works with cibuildwheel, so I think it's an issue with the action.

**Comment by messense** [comment]
Can you try install them via setup-python action before calling this action?

**Comment by henryiii** [comment]
Yes, I tried that previously, but I can repeat it in the PR, just a sec.

**Comment by henryiii** [comment]
Same: https://github.com/tox-dev/pyproject-fmt-rust/actions/runs/9058486065/job/24884203304?pr=11

**Comment by messense** [comment]
Probably something has changed in github actions tool cache for PyPy.

https://github.com/PyO3/maturin-action/blob/b49dce4c3301c4291fbfb8519934c34dedf502ac/src/index.ts#L769-L779

**Comment by messense** [comment]
In the meantime you can try enable `generate-import-lib` feature of pyo3.

See https://pyo3.rs/v0.21.2/features.html?highlight=generate-import-lib#generate-import-lib

**Comment by henryiii** [comment]
It's in a `PyPy` folder: https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#hosted-tool-cache

**Comment by henryiii** [comment]
Assuming this the right way to enable it:

```toml
[dependencies]
pyo3 = { version = "0.21.2", features = ["abi3-py38", "generate-import-lib"] }
```

This cause Maturin to simply ignore the `pypy` jobs completely and only produce the main wheel.  The jobs "passes", but only one wheel is produced. Interestingly, the unix jobs still seem fine and produce PyPy wheels.

**Comment by messense** [comment]
Interesting, might be a bug in maturin, I'll take a look when I have the time.

**Comment by henryiii** [comment]
Is there a way for me to see the core.info printouts? I'm still not seeing it with the recent update.

**Comment by messense** [comment]
core.info should be visible by default, so it's simply didn't run.

https://github.com/tox-dev/pyproject-fmt-rust/actions/runs/9068434736/job/24915732974?pr=11#step:3:34 this one does have the output.

**Comment by rominf** [comment]
I can confirm that the issue is still actual. Unfortunately, specifying the interpreter manually via `--interpreter` does not help either. I thought it was an issue with the `setup-python` GitHub Action, so I created two PRs in my fork of `setup-python`: 

- https://github.com/rominf/setup-python/pull/2 – tests for building Python extensions (C extension using setuptools and Rust extension using pyo3 and maturin).
- https://github.com/rominf/setup-python/pull/1 – setting the `LIB` environmental variable after installing PyPy on Windows (on top of the PR above). 

As you might see, `link.exe` fails to find `python*.lib` on Windows with PyPy interpreter requested and setting `LIB` helps `maturin` to build the extension. However, I tend to believe that the issue is better solved in `maturin` or the `maturin` GitHub Action.

I am open to providing more information or filling another issue if needed.
